framework/src/outputs/Console.C

```diff
@@ -251,6 +236,24 @@ Console::~Console()
 void
 Console::initialSetup()
 {
+  // If --timing was used from the command-line, do nothing, all logs are enabled
+  // Also, only allow the main app to change the perf_log settings.
+  if (!_timing && _app.name() == "main")
+  {
+    if (_perf_log || _setup_log || _solve_log || _perf_header || _setup_log_early)
+      _app.setLoggingRequested();
+
+    // Disable performance logging if nobody needs logging
+    if (!_app.getLoggingRequested())
+      Moose::perf_log.disable_logging();
+
+    // Disable libMesh log
+#ifdef LIBMESH_ENABLE_PERFORMANCE_LOGGING
```

Comment:
I realize you didn't add this code block in this PR, but I don't think it needs to be wrapped in the `ifdef`.  The global `perflog` object exists no matter what, the `ifdef` only controls the behavior of the `START_LOG`, `STOP_LOG`, etc. macros.